src/box/lua/upgrade.lua

```diff
@@ -2125,6 +2126,10 @@ local function downgrade_impl(version_str, dry_run)
     end
 
     local schema_version_cur = get_version()
+    local app_version = tarantool.version:match('^%d+%.%d+%.%d+')
+    if schema_version_cur > mkversion.parse(app_version) then
+        error('Schema version is more recent than application.')
```

Comment:
Nit: Let's make the error message more verbose:

```lua
"Cannot downgrade as current schema version %s is newer than Tarantool version %s"
```

Reply:
Done.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/lua/upgrade.lua b/src/box/lua/upgrade.lua
index e92bb6196..04463cfe6 100644
--- a/src/box/lua/upgrade.lua
+++ b/src/box/lua/upgrade.lua
@@ -2128,7 +2128,9 @@ local function downgrade_impl(version_str, dry_run)
     local schema_version_cur = get_version()
     local app_version = tarantool.version:match('^%d+%.%d+%.%d+')
     if schema_version_cur > mkversion.parse(app_version) then
-        error('Schema version is more recent than application.')
+        local err = "Cannot downgrade as current schema version %s is newer" ..
+                    " than Tarantool version %s"
+        error(err:format(schema_version_cur, app_version))
     end
     local schema_version_dst = app2schema_version(version)
     if schema_version_cur < schema_version_dst then
diff --git a/test/box-luatest/downgrade_test.lua b/test/box-luatest/downgrade_test.lua
index b805d96a6..8b3031539 100644
--- a/test/box-luatest/downgrade_test.lua
+++ b/test/box-luatest/downgrade_test.lua
@@ -89,9 +89,10 @@ g.test_downgrade_from_more_recent_version = function(cg)
         box.space._schema:replace({'version', tonumber(major), tonumber(minor),
                                    tonumber(patch) + 1})
         local new_version = schema_version()
-        t.assert_error_msg_contains(
-            'Schema version is more recent than application.',
-            box.schema.downgrade, app_version)
+        local err = 'Cannot downgrade as current schema version %s is newer' ..
+                    ' than Tarantool version %s'
+        t.assert_error_msg_contains(err:format(new_version, app_version),
+                                    box.schema.downgrade, app_version)
         t.assert_equals(schema_version(), new_version)
     end)
 end
```

</details>